homeassistant/components/hassio/http.py

```diff
@@ -124,7 +124,7 @@ async def _command_proxy(
 def _init_header(request: web.Request) -> dict[str, str]:
     """Create initial header."""
     headers = {
-        X_HASSIO: os.environ.get("HASSIO_TOKEN", ""),
+        X_HASSIO: os.environ.get("SUPERVISOR_TOKEN", ""),
```

Comment:
`X-Hassio-Key` is also deprecated and should be replaced as well, but that deserves a separate PR.